core/src/impl/Kokkos_Spinwait.hpp

```diff
@@ -38,8 +38,8 @@ enum class WaitMode : int {
 void host_thread_yield(const uint32_t i, const WaitMode mode);
 
 template <typename T>
-std::enable_if_t<std::is_integral<T>::value, void> root_spinwait_while_equal(
-    T const volatile& flag, const T value) {
+std::enable_if_t<std::is_integral_v<T> || std::is_enum_v<T>, void>
+root_spinwait_while_equal(T const volatile& flag, const T value) {
```

Comment:
Why are these templated at all?  Was it some way of avoiding circular dependencies?

Reply:
Yeah but now that I think about it more, I could also move the `enum` outside of the class into a separate file, move `Kokkos_Spinwait` to the `Threads` backend folder, and remove the template parameter. Not sure if one solution is better than the other.